lib/git.js

```diff
@@ -427,14 +428,30 @@ class Git extends EventEmitter {
    * @method listen
    * @memberof Git
    * @param  {Number}   port     - the port to start the server on
+   * @param  {Array}   port     - the https options
    * @param  {Function} callback - the function to call when server is started or error has occured
    */
-  listen(port, callback) {
-      var self = this;
-      this.server = http.createServer(function(req, res) {
+  listen(port, options, callback) {
+    if(typeof options == 'function') {
+      callback = options;
+    }
+
+    var self = this;
+    this.server = http.createServer(function(req, res) {
+        self.handle(req, res);
+    });
+    this.server.listen(port, function() {
+      callback(port, 'http');
+    });
+
+    if(typeof options != 'function') {
+      this.server = https.createServer(options, function(req, res) {
```

Comment:
the omission of options makes it so that the user would want http? Could an option not be `type: https|http`?

Reply:
Would You Be Fine If I Was To Do The Following:
'listen(options, callback) {'
So That The Input For The Listen Function is What I Quoted Above

Were Options Is An Array Including The HTTP Port, HTTPS Port, HTTPS Cert and Key And Also An Option To Disable HTTP (And Disable HTTPS But Not Including The Cert, Key Or Port.

Reply:
If Not My Last Comment Would You Like It To Be More Like
'listen(port, type, options, callback) {'

Also With options only being defined if the type is https. Type Being A Variable Defining If You HTTP or HTTPS (This Would Them Require Two Listen Statements To Start A HTTPS And HTTP Server?